src/main/java/EventCommand.java

```diff
@@ -0,0 +1,31 @@
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
+public class EventCommand extends Command {
+    private String description;
+    private String at;
+
+    public EventCommand(String action, String description, String at) {
+        super(action);
+        this.description = description;
+        this.at = at;
+    }
+
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws CommandException {
+        try {
+            StringBuilder str = new StringBuilder();
```

Comment:
Could you think of more specific and descriptive names for your variables? "str", "temp" and "current" may be vague at the first glance :)

Reply:
I'm thinking about whether to keep these scratch variables short as they are not returned, and are not used outside 2 -3 lines of code.